mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -407,6 +449,60 @@ private[classification] class MultiClassSummarizer extends Serializable {
   }
 }
 
+@Experimental
+class LogisticRegressionTrainingSummary private[classification] (
+    predictions: DataFrame,
+    probabilityCol: String,
+    labelCol: String,
+    val objectiveHistory: Array[Double])
+  extends LogisticRegressionSummary(predictions, probabilityCol, labelCol) {
+
+  /** Number of training iterations until termination */
+  val totalIterations = objectiveHistory.length
+
+}
+
+@Experimental
+class LogisticRegressionSummary private[classification] (
+  @transient val predictions: DataFrame,
+  val probabilityCol: String,
+  val labelCol: String) extends Serializable {
+
+  @transient val metrics = new BinaryClassificationMetrics(
+    predictions.select(probabilityCol, labelCol).map {
+      case Row(score: Vector, label: Double) => (score(1), label)
+    }
+  )
+
+  /**
+   * Returns the receiver operating characteristic (ROC) curve,
+   * which is an RDD of (false positive rate, true positive rate)
+   * with (0.0, 0.0) prepended and (1.0, 1.0) appended to it.
+   */
+  val roc: RDD[(Double, Double)] = metrics.roc()
```

Comment:
It seems that the size of the ROC curve and all other metrics is equal to the size of the data. (i.e it chooses every possible score as a threshold)